test/gateway/eip20_gateway/confirm_redeem_intent.js

```diff
@@ -0,0 +1,494 @@
+
+    messageHash = TestDataJSON.gateway.confirm_redeem_intent.return_value.returned_value.messageHash_;
+    assert.strictEqual(
+      result,
+      messageHash,
+      `Message hash from event must be equal to ${messageHash}.`,
+    );
+
+  });
+
+  it('should emit `RedeemIntentConfirmed` event.', async function () {
+
+    let tx = await eip20Gateway.confirmRedeemIntent(
+      redeemRequest.redeemer,
+      redeemRequest.nonce,
+      redeemRequest.beneficiary,
+      redeemRequest.amount,
+      redeemRequest.gasPrice,
+      redeemRequest.gasLimit,
+      redeemRequest.blockNumber,
+      redeemRequest.hashLock,
+      redeemRequest.storageProof,
+    );
+
+    let event = EventDecoder.getEvents(tx, eip20Gateway);
+
+    assert.isDefined(
+      event.RedeemIntentConfirmed,
+      'Event RedeemIntentConfirmed must be emitted.',
+    );
+
+    let eventData = event.RedeemIntentConfirmed;
+
+    let params = TestDataJSON.gateway.confirm_redeem_intent.params;
+
+    assert.strictEqual(
+      eventData._messageHash,
+      messageHash,
+      `Message hash from event must be equal to ${messageHash}.`,
+    );
+
+    assert.strictEqual(
+      eventData._redeemer,
+      params.redeemer,
+      `Redeemer address ${eventData._redeemer} from event must be equal to ${params.redeemer}.`,
+    );
+
+    let nonce = new BN(params.nonce, 16);
+    assert.strictEqual(
+      nonce.eq(eventData._redeemerNonce),
+      true,
+      `Nonce value ${eventData._redeemerNonce} from event must be equal to ${nonce.toString(10)}.`,
+    );
+
+    assert.strictEqual(
+      eventData._beneficiary,
+      params.beneficiary,
+      `Beneficiary address from event must be equal to ${params.beneficiary}.`,
+    );
+
+    let amount = new BN(params.amount, 16);
+    assert.strictEqual(
+      amount.eq(eventData._amount),
+      true,
+      `Amount ${eventData._amount} from event must be equal to ${amount.toString(10)}.`,
+    );
+
+    let blockHeight = new BN(params.blockNumber, 16);
+    assert.strictEqual(
+      blockHeight.eq(eventData._blockHeight),
+      true,
+      `Block height ${eventData._blockHeight} from event must be equal to ${blockHeight.toString(10)}.`,
+    );
+
+    assert.strictEqual(
+      eventData._hashLock,
+      params.hashLock,
+      `Hash lock from event must be equal to ${params.hashLock}.`,
+    );
+
+  });
+
+  it('should confirm new redeem intent if status of previous ' +
+    'confirmed redeem message is revoked', async function () {
+
+      await eip20Gateway.confirmRedeemIntent(
+        redeemRequest.redeemer,
+        redeemRequest.nonce,
+        redeemRequest.beneficiary,
+        redeemRequest.amount,
+        redeemRequest.gasPrice,
+        redeemRequest.gasLimit,
+        redeemRequest.blockNumber,
+        redeemRequest.hashLock,
+        redeemRequest.storageProof,
+      );
+
+      await eip20Gateway.setInboxStatus(
+        TestDataJSON.gateway.confirm_redeem_intent.return_value.returned_value.messageHash_,
```

Comment:
We can store `messageHash` in `redeemRequest` and use it wherever its needed in the test.